source/MaterialXGenShader/Syntax.cpp

```diff
@@ -246,4 +276,38 @@ string AggregateTypeSyntax::getValue(const Value& value, bool /*uniform*/) const
     return valueString.empty() ? valueString : getName() + "(" + valueString + ")";
 }
 
+StructTypeSyntax::StructTypeSyntax(const Syntax* parentSyntax, const string& name, const string& defaultValue, const string& uniformDefaultValue,
+                                   const string& typeAlias, const string& typeDefinition, const StringVec& members) :
+    TypeSyntax(name, defaultValue, uniformDefaultValue, typeAlias, typeDefinition, members), _parentSyntax(parentSyntax)
+{
+}
+
+string StructTypeSyntax::getValue(const Value& value, bool /*uniform*/) const
+{
+    const AggregateValue& aggValue = static_cast<const AggregateValue&>(value);
+
+    auto typeDesc = TypeDesc::get(aggValue.getTypeString());
+    auto structTypeDesc = StructTypeDesc::get(typeDesc.getStructIndex());
+
+    string result = "{";
+
+    string separator = "";
+    for (const auto& memberValue : aggValue.getMembers()) {
+        result += separator;
+        separator = ";";
+
+        auto memberTypeName = memberValue->getTypeString();
+        auto memberTypeDesc = TypeDesc::get(memberTypeName);
+
+        // recursively use the syntax to generate the output, so we can supported nested structs.
```

Comment:
For consistency, we should capitalize the first letter of each sentence in comments.